### PR TITLE
[WIP] Removed `ProductDiscretePSpace` and `ProductContinuousPSpace`

### PR DESCRIPTION
**Pull request by akash9712**
#### Brief description of what is fixed or changed
Removed classes `ProductDiscretePSpace` and `ProductContinuousPSpace` and moved the methods to `ProductPSpace`. No new tests have been added, as tests already present check for the functionality of these classes. 

#### Other comments
closes #14730

### CODE REVIEW

File: sympy/core/tests/test_args.py [hunk 1]
```diff
@@ -761,7 +760,6 @@ def test_sympy__stats__crv__ConditionalContinuousDomain():
     D = SingleContinuousDomain(x, Interval(-oo, oo))
     assert _test_args(ConditionalContinuousDomain(D, x > 0))
 
-
```
Comment (Upabjojr) [inline sympy/core/tests/test_args.py]: we usually leave 2 newlines on the base scope

File: sympy/stats/tests/test_discrete_rv.py [hunk 2]
```diff
@@ -86,3 +88,16 @@ def test_conditional():
...
+    X2 = Geometric('X2', S(1)/3)
+    from sympy import symbols
+    n = symbols('n')
+    oo = S.Infinity
+    assert str(P(X1 + X2 < 3, evaluate=False) == Sum(Piecewise((2**(X2 - n - 2)*(2/3)**(X2 - 1)/6,
+    (-X2 + n + 3 >= 1) & (-X2 + n + 3 < oo)), (0, True)), (X2, 1, oo), (n, -oo, -1)))
+    assert str(P(X1 + X2 > 3) == Sum(Piecewise((2**(X2 - n - 2)*(2/3)**(X2 - 1)/6,
+        (-X2 + n + 3 >= 1) & (-X2 + n + 3 < oo)), (0, True)), (X2, 1, oo), (n, 1, oo)))
+    assert str(P(Eq(X1 + X2, 3)) == Sum(
+        Piecewise((2**(X2 - 2)*(S(2)/3)**(X2 - 1)/6, X2 <= 2), (0, True)), (X2, 1, oo)))
```
Comment (Upabjojr) [inline sympy/stats/tests/test_discrete_rv.py]: I was checking this result with Mathematica, I get:
```
In[12]:= Probability[x+y==3, {x\[Distributed]GeometricDistribution[1/2], y\[Distributed] GeometricDistribution[1/3]}]
         175
Out[12]= ----
         1296
```
I think your expression is equal to `7/36`

Comment (Upabjojr) [reply sympy/stats/tests/test_discrete_rv.py]: Sorry, Mathematica and SymPy have slightly different definitions of the PDF.

**Comment by Upabjojr** [comment]
What about mixed discrete and continuous product spaces?

**Comment by akash9712** [comment]
If the expression contains both discrete and continuous random variables, the product space is taken to be continuous, with a discrete component. The density is then calculated using `DiracDelta`, as on [L376](https://github.com/sympy/sympy/pull/14777/files#diff-650b9a2935441553d2e613dbf85140e8R376).

**Comment by Upabjojr** [comment]
OK, can you add a test for the mixed case?

File: sympy/stats/tests/test_discrete_rv.py [hunk 3]
```diff
@@ -86,3 +88,16 @@ def test_conditional():
...
     assert P(Y > 2, Y < 2) == 0
     assert P(Eq(Y, 3), Y >= 0) == 9*exp(-3)/2
+
+def test_product_spaces():
+    X1 = Geometric('X1', S(1)/2)
+    X2 = Geometric('X2', S(1)/3)
+    from sympy import symbols
+    n = symbols('n')
+    oo = S.Infinity
+    assert str(P(X1 + X2 < 3, evaluate=False) == Sum(Piecewise((2**(X2 - n - 2)*(2/3)**(X2 - 1)/6,
```
Comment (Upabjojr) [inline sympy/stats/tests/test_discrete_rv.py]: this assert usage is wrong, you are evaluating a string.

Comment (Upabjojr) [reply sympy/stats/tests/test_discrete_rv.py]: I would suggest `str(P( ... )) == """ string expression """`

File: sympy/stats/tests/test_mix.py [hunk 4]
```diff
@@ -10,3 +10,12 @@ def test_density():
     X = Poisson(x, rate)
     assert isinstance(pspace(X), ProductPSpace)
     assert density(X, Eq(rate, rate.symbol)) == PoissonDistribution(l)
+
+def test_mix_expression():
+    Y, E = Poisson('Y', 1), Exponential('E', 1)
+    assert P(Eq(Y + E, 1)) == 0
+    assert P(Ne(Y + E, 2)) == 1
+    assert str(P(E + Y < 2, evaluate=False)) == """Integral(Sum(exp(-1)*Integral"""\
++"""(exp(-E)*DiracDelta(-_z + E + Y - 2), (E, 0, oo))/factorial(Y), (Y, 0, oo)), (_z, -oo, 0))"""
```
Comment (Upabjojr) [inline sympy/stats/tests/test_mix.py]: I believe this integral diverges to infinity.

**Pull request merged**